### ISSUE TITLE
Unable to start due to uninitialized constant

### ISSUE THREAD

**Issue by tao-oat** [Shopify/tainted_love#3]
After following the setup instructions in the README exactly (for a Rails project), I get the following error when I run any rails command:
```
NameError: uninitialized constant TaintedLove::Replacer::Base
```
This is with Rails 5.2.3 and Ruby 2.6.1.

**Comment by Becojo** [comment]
I can't reproduce on Mac with the versions given and a basic scaffold.

- What platform is the application running? 
- Can you share parts of the stack trace to know where this error is raised?

I'm thinking it's this hack that doesn't load files in the same order   https://github.com/Shopify/tainted_love/blob/b38c1de7b0d53cfe7f54ee1c838e0bab598e890a/lib/tainted_love.rb

**Comment by david-a-wheeler** [comment]
I have exactly the same problem. I tried to add `tainted_love` to the CII Best Practices badge's test suite. Its code can be downloaded here: https://github.com/coreinfrastructure/best-practices-badge

I tried to run this on Ubuntu (VERSION="16.04.6 LTS (Xenial Xerus)"), using "rails test" which uses minitest. We use rbenv. Its output is:

~~~~
Coverage report generated for MiniTest to /home/dwheeler/best-practices-badge/coverage. 0 / 2999 LOC (0.0%) covered.
Traceback (most recent call last):
	33: from bin/rails:6:in `<main>'
	32: from bin/rails:6:in `require'
	31: from /home/dwheeler/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/railties-5.2.2.1/lib/rails/commands.rb:18:in `<top (required)>'
	30: from /home/dwheeler/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/railties-5.2.2.1/lib/rails/command.rb:46:in `invoke'
	29: from /home/dwheeler/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/railties-5.2.2.1/lib/rails/command/base.rb:65:in `perform'
	28: from /home/dwheeler/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	27: from /home/dwheeler/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	26: from /home/dwheeler/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	25: from /home/dwheeler/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/railties-5.2.2.1/lib/rails/commands/test/test_command.rb:33:in `perform'
	24: from /home/dwheeler/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/railties-5.2.2.1/lib/rails/test_unit/runner.rb:39:in `run'
	23: from /home/dwheeler/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/railties-5.2.2.1/lib/rails/test_unit/runner.rb:50:in `load_tests'
	22: from /home/dwheeler/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/railties-5.2.2.1/lib/rails/test_unit/runner.rb:50:in `each'
	21: from /home/dwheeler/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/railties-5.2.2.1/lib/rails/test_unit/runner.rb:50:in `block in load_tests'
	20: from /home/dwheeler/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/railties-5.2.2.1/lib/rails/test_unit/runner.rb:50:in `require'
	19: from /home/dwheeler/best-practices-badge/test/capybara_feature_test.rb:8:in `<top (required)>'
	18: from /home/dwheeler/best-practices-badge/test/capybara_feature_test.rb:8:in `require'
	17: from /home/dwheeler/best-practices-badge/test/test_helper.rb:51:in `<top (required)>'
	16: from /home/dwheeler/best-practices-badge/test/test_helper.rb:51:in `require'
	15: from /home/dwheeler/best-practices-badge/config/environment.rb:8:in `<top (required)>'
	14: from /home/dwheeler/best-practices-badge/config/environment.rb:8:in `require_relative'
	13: from /home/dwheeler/best-practices-badge/config/application.rb:13:in `<top (required)>'
	12: from /home/dwheeler/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/bundler-2.1.4/lib/bundler.rb:174:in `require'
	11: from /home/dwheeler/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/bundler-2.1.4/lib/bundler/runtime.rb:58:in `require'
	10: from /home/dwheeler/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/bundler-2.1.4/lib/bundler/runtime.rb:58:in `each'
	 9: from /home/dwheeler/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/bundler-2.1.4/lib/bundler/runtime.rb:69:in `block in require'
	 8: from /home/dwheeler/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/bundler-2.1.4/lib/bundler/runtime.rb:69:in `each'
	 7: from /home/dwheeler/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/bundler-2.1.4/lib/bundler/runtime.rb:74:in `block (2 levels) in require'
	 6: from /home/dwheeler/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/bundler-2.1.4/lib/bundler/runtime.rb:74:in `require'
	 5: from /home/dwheeler/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/tainted_love-0.4.0/lib/tainted_love.rb:3:in `<top (required)>'
	 4: from /home/dwheeler/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/tainted_love-0.4.0/lib/tainted_love.rb:3:in `each'
	 3: from /home/dwheeler/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/tainted_love-0.4.0/lib/tainted_love.rb:3:in `block in <top (required)>'
	 2: from /home/dwheeler/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/tainted_love-0.4.0/lib/tainted_love/validator/action_dispatch_diagnostics.rb:3:in `<top (required)>'
	 1: from /home/dwheeler/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/tainted_love-0.4.0/lib/tainted_love/validator/action_dispatch_diagnostics.rb:4:in `<module:TaintedLove>'
/home/dwheeler/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/tainted_love-0.4.0/lib/tainted_love/validator/action_dispatch_diagnostics.rb:5:in `<module:Validator>': uninitialized constant TaintedLove::Validator::Base (NameError)
~~~~

Let me know if there's any other information I can provide. Otherwise, good luck! It looks interesting.

**Comment by Becojo** [comment]
@david-a-wheeler Thanks for the information. I took a look at the application and I was able to run your test suite with TaintedLove. FYI, this tool is very experimental. You probably don't want this enabled in your test suite all the time. 

I made a change to address the issue in 0.4.1. I also added a note about in the readme about Ruby 2.7 removing taint checking and as such there is currently no active development of this project.
